embed-code-go/embed_code.go

```diff
@@ -0,0 +1,177 @@
+// Copyright 2024, TeamDev. All rights reserved.
+//
+// Redistribution and use in source and/or binary forms, with or without
+// modification, must retain the above copyright notice and the following
+// disclaimer.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+// "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+// LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+// A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+// OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+// LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+// DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+// THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+package main
+
+import (
+	"embed-code/embed-code-go/cli"
+	"embed-code/embed-code-go/configuration"
+	"flag"
+	"fmt"
+
+	"os"
+
+	"gopkg.in/yaml.v3"
+)
+
+// Struct with user-specified flags.
+//
+// codeRoot — a path to a root directory with code files.
+//
+// docsRoot — a path to a root directory with docs files.
+//
+// configPath — a path to a yaml configuration file which contains the roots.
+//
+// checkUpToDate — true to check for code embeddings to be up-to-date. Otherwise, the embedding is performed.
+type flags struct {
+	codeRoot      string
+	docsRoot      string
+	configPath    string
+	checkUpToDate bool
+}
+
+// Struct with roots that contained in a yaml configuration file.
+//
+// codeRoot — a path to a root directory with code files.
+//
+// docsRoot — a path to a root directory with docs files.
+type configFields struct {
+	CodeRoot string `yaml:"code_root"`
+	DocsRoot string `yaml:"docs_root"`
+}
+
+// Reads the roots from the provided configPath and returns a configFields struct.
+//
+// configPath — a path to a yaml configuration file which contains the roots.
+//
+// Returns a configFields struct filled with the roots.
+func readRootsFromConfig(configPath string) configFields {
+	content, err := os.ReadFile(configPath)
+	if err != nil {
+		panic(err)
+	}
+
+	configFields := configFields{}
+	err = yaml.Unmarshal(content, &configFields)
+	if err != nil {
+		panic(err)
+	}
+
+	return configFields
+}
+
+// Reads user-specified flags from the command line.
+//
+// Returns a flags struct filled with the corresponding flags.
+func readFlags() flags {
+	codeRoot := flag.String("code_root", "", "a path to a root directory with code files")
+	docsRoot := flag.String("docs_root", "", "a path to a root directory with docs files")
+
+	configPath := flag.String("config_path", "",
+		"a path to a yaml configuration file, which contains 'code_root' and 'docs_root' fields")
```

Comment:
Ideally, we want to be shorter and more precise:
```
a path to a YAML configuration file defining 'code_root' and 'docs_root'
```

In general, I would not mention specific values that we look for. They may change, there may be more of them, etc. And this piece will get outdated soon.